### PR TITLE
Add findbugs and fix findbugs

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

File: src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java [hunk 1]
```diff
@@ -24,16 +26,17 @@
...
  * <p>
- * In comparison to plain {@code ThreadLocal}, this implementation:<ul>
- * <li>from the point of view of a single thread,
- * each thread-local
- * {code #get} requires access to four objects instead of two
- * <li>is fractionally slower in terms of CPU cycles for {code #get}
- * <li>uses around twice the memory for each thead-local value
- * <li>uses around four times the memory for each {@code ThreadLocal}
- * <li>may release thread-local values for garbage collection more promptly
+ * In comparison to plain {@code ThreadLocal}, this implementation:
```
Comment (ChrisGreenaway) [inline src/main/java/org/terracotta/statistics/util/VicariousThreadLocal.java]: Can you undo this spurious doc diff

File: src/main/java/org/terracotta/statistics/extended/CompoundOperationImpl.java [hunk 2]
```diff
@@ -230,7 +230,7 @@ public boolean expire(long expiryTime) {
     } else {
       boolean expired = true;
       for (ResultImpl<?> o : operations.values()) {
-        expired &= o.expire(expiryTime);
+        expired = expired && o.expire(expiryTime);
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/extended/CompoundOperationImpl.java]: Pretty sure this needs to be a non short-circuiting logical and operation.

File: src/main/java/org/terracotta/statistics/extended/CompoundOperationImpl.java [hunk 3]
```diff
@@ -242,7 +242,7 @@ public boolean expire(long expiryTime) {
           it.remove();
         }
       }
-      return expired & compounds.isEmpty() & ratios.isEmpty();
+      return expired && compounds.isEmpty() && ratios.isEmpty();
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/extended/CompoundOperationImpl.java]: This one is good.

File: src/main/java/org/terracotta/statistics/extended/ResultImpl.java [hunk 4]
```diff
@@ -92,7 +92,7 @@ void start() {
    * @return true, if successful
    */
   boolean expire(long expiryTime) {
-    return (count.expire(expiryTime) & rate.expire(expiryTime) & latency.expire(expiryTime));
+    return count.expire(expiryTime) && rate.expire(expiryTime) && latency.expire(expiryTime);
```
Comment (chrisdennis) [inline src/main/java/org/terracotta/statistics/extended/ResultImpl.java]: Think this needs to be non-short-circuiting too.

**Pull request merged**